### PR TITLE
Add MonadThrow laws

### PR DESCRIPTION
**Pull request by danimontoya**
Covers small part of #2029 .

Create test suite for MonadThrowLaws.

### CODE REVIEW

File: modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadThrowLaws.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import arrow.typeclasses.EqK
+import arrow.typeclasses.Functor
+import arrow.typeclasses.MonadThrow
+import arrow.typeclasses.Selective
+import io.kotlintest.fail
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import io.kotlintest.shouldThrowAny
+
+object MonadThrowLaws {
```
Comment (aballano) [inline modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadThrowLaws.kt]: I believe you should probably hook this up in `MonadDeferLaws`'s laws functions as:

```
BracketLaws.laws(...) +
  MonadThrowLaws.laws(...) +
      monadDeferLaws(...)
``` 

Although I'm unsure if this won't run all MonadError tests twice as both Bracket and MonadThrow come from MonadError, but independently

Comment (danimontoya) [reply modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadThrowLaws.kt]: yeah, I guess that should be aligned so it doesn't execute the tests more times than required

Comment (aballano) [reply modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadThrowLaws.kt]: Ah nevermind, I just noticed that in `testLaws` function there's a distinct! So good to go with the suggested change

Comment (danimontoya) [reply modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadThrowLaws.kt]: perfect then

**Pull request merged**